src/logic/safe/store/reducer/safe.ts

```diff
@@ -51,7 +51,7 @@ export default handleActions(
       return state.updateIn(
         ['safes', safeAddress],
         makeSafe({ name: 'LOADED SAFE', address: safeAddress }),
-        (prevSafe) => prevSafe.merge(safe),
+        (prevSafe) => prevSafe.mergeDeep(safe),
```

Comment:
This is not fixing the problem. For me the tokens disappear on the first load, displaying Ether alone. Then after few seconds they appear again.

The mergeDeep is virtually fixing this if applied in line 80 [ADD_SAFE], because we are calling [ADD_SAFE] again after the app is initialized and this is deleting the token values,having to wait until they load again.
I say that this virtually fix it because it introduces another issue. If mergeDeep is applied in [ADD_SAFE] it will add the owners again and again on each reload making the list of owners grow on the same basis as the number of owners we have. They will be shown duplicate when signing a transaction or in settings section.

Reply:
@dasanra please take a look again
I changed the ADD_SAFE action to return the existing state and not modify it if the safe already exists. To solve it in a fundamental way, we would have to rethink how safe is loaded when you land on the safe page as there is clearly a concurrency problem between add/update safe actions